backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/RemoveImageCommandCallback.java

```diff
@@ -0,0 +1,82 @@
+package org.ovirt.engine.core.bll.storage.disk.image;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.enterprise.inject.Typed;
+import javax.inject.Inject;
+
+import org.ovirt.engine.core.bll.tasks.CommandCoordinatorUtil;
+import org.ovirt.engine.core.bll.tasks.interfaces.CommandCallback;
+import org.ovirt.engine.core.common.action.RemoveImageParameters;
+import org.ovirt.engine.core.common.businessentities.AsyncTaskStatus;
+import org.ovirt.engine.core.compat.CommandStatus;
+import org.ovirt.engine.core.compat.Guid;
+import org.ovirt.engine.core.vdsbroker.vdsbroker.VDSErrorException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Typed(RemoveImageCommandCallback.class)
+public class RemoveImageCommandCallback implements CommandCallback {
+    private static final Logger log = LoggerFactory.getLogger(RemoveImageCommandCallback.class);
+
+    @Inject
+    private CommandCoordinatorUtil commandCoordinatorUtil;
+
+    @Override
+    public void doPolling(Guid cmdId, List<Guid> childCmdIds) {
+        RemoveImageCommand<RemoveImageParameters> command = getRemoveImageCommand(cmdId);
+        Set<Guid> taskIds = new HashSet<>(command.getReturnValue().getVdsmTaskIdList());
+        Map<Guid, AsyncTaskStatus> idToTaskStatusMap = commandCoordinatorUtil.pollTasks(taskIds);
+        for (Map.Entry<Guid, AsyncTaskStatus> idToTaskStatus : idToTaskStatusMap.entrySet()) {
+            Guid id = idToTaskStatus.getKey();
+            AsyncTaskStatus status = idToTaskStatus.getValue();
+            if (status.getTaskIsRunning()) {
+                log.info("Waiting on remove image command to complete the task (taskId = '{}')", id);
+                return;
+            }
+        }
+
+        List<AsyncTaskStatus> failedTasks = new ArrayList<>();
+        idToTaskStatusMap.forEach((id, status) -> {
+            if (!status.getTaskEndedSuccessfully()) {
+                failedTasks.add(status);
+             }
+        });
+        if (failedTasks.isEmpty()) {
+            command.setSucceeded(true);
+            command.setCommandStatus(CommandStatus.SUCCEEDED);
+            command.persistCommand(command.getParameters().getParentCommand());
+            log.info("Remove image command has completed successfully for image '{}' with async task '{}'.",
```

Comment:
For `task` you print the `Set` of task IDs, might be more than 1 task ID? If so, need to to update the message?

Reply:
Currently there is one task. But the infra is built to track multiple tasks, so I figured it would be better to leave it this way for it not to break if someone adds a new task in the command. It's the same in DestroyImage btw

Reply:
Ack, can resolve the comment.